hathor/transaction/base_transaction.py

```diff
@@ -742,6 +743,51 @@ def _update_feature_activation_bit_counts(self) -> None:
         # This method lazily calculates and stores the value in metadata
         self.get_feature_activation_bit_counts()
 
+    def _update_closest_block_metadata(self) -> None:
+        """
+        Set the tx's closest_block metadata.
+        For blocks, it's always None. For Transactions, it's the Block with the greatest height that is a direct
+        or indirect dependency (ancestor) of the transaction, including both funds and confirmation DAGs.
+        It's calculated by propagating the metadata forward in the DAG,
+        and it's used by Feature Activation for Transactions.
+        """
+        from hathor.transaction import Block, Transaction
+        if isinstance(self, Block):
+            return
+        assert isinstance(self, Transaction)
+        assert self.storage is not None
+        metadata = self.get_metadata()
+
+        if self.is_genesis:
+            metadata.closest_block = self._settings.GENESIS_BLOCK_HASH
+            return
+
+        closest_block: Block | None = None
+        dependency_ids = self.parents + [tx_input.tx_id for tx_input in self.inputs]
+
+        for vertex_id in dependency_ids:
+            vertex = self.storage.get_transaction(vertex_id)
+            vertex_meta = vertex.get_metadata()
+            this_closest_block: Block
+
+            if isinstance(vertex, Block):
+                assert vertex_meta.closest_block is None
+                this_closest_block = vertex
+            elif isinstance(vertex, Transaction):
+                this_closest_block_id = (
+                    self._settings.GENESIS_BLOCK_HASH if vertex.is_genesis else not_none(vertex_meta.closest_block)
```

Comment:
I guess you can safely do `not_none(vertex_meta.closest_block)`.

Reply:
Done!

Reply:
Thanks!